src/sharepoint/rest/items.ts

```diff
@@ -1,42 +1,186 @@
 
-    constructor(url: string | Queryable) {
-        super(url, "items");
+    /**
+     * Creates a new instance of the Items class
+     * 
+     * @param baseUrl The url or Queryable which forms the parent of this fields collection
+     */
+    constructor(baseUrl: string | Queryable) {
+        super(baseUrl, "items");
     }
 
+    /**
+     * Gets an Item by id
+     * 
+     * @param id The integer id of the item to retrieve
+     */
     public getById(id: number): Item {
         this.concat(`(${id})`);
-        return this;
+        return new Item(this);
     }
 
-    // gettable stub for mixin
+    /**
+     * Execute the get request
+     * 
+     */
     public get(): Promise<any> { return; }
 
-    // selectable stub for mixin
+    /**
+     * Select the fields to return
+     * 
+     * @param selects One or more fields to return
+     */
     public select(...selects: string[]): Items { return; }
 
-    // filterable stub for mixin
+    /**
+     * Applies a filter to the request
+     * 
+     * @param filter The filter string (docs: https://msdn.microsoft.com/en-us/library/office/fp142385.aspx)
+     */
     public filter(filter: string): Items { return; }
 }
-
 Util.applyMixins(Items, Mixins.Gettable, Mixins.Selectable, Mixins.Filterable);
 
+
+/**
+ * Descrines a single Item instance
+ * 
+ */
 export class Item extends Queryable implements Mixins.Gettable, Mixins.Selectable {
 
+    /**
+     * Creates a new instance of the Items class
+     * 
+     * @param baseUrl The url or Queryable which forms the parent of this fields collection
+     */
     constructor(baseUrl: string | Queryable) {
         super(baseUrl);
     }
 
-    // gettable stub for mixin
+    /**
+     * Gets the set of attachments for this item
+     * 
+     */
+    public get attachmentFiles(): FilterableSelectableGettable<any> {
+        this.append("AttachmentFiles");
+        return new FilterableSelectableGettable<any>(this);
+    }
+
+    /**
+     * Gets the content type for this item
+     * 
+     */
+    public get contentType(): ContentType {
+        this.append("ContentType");
+        return new ContentType(this);
+    }
+
+    /**
+     * Gets the effective base permissions for the item
+     * 
+     */
+    public get effectiveBasePermissions(): Gettable<any> {
+        this.append("EffectiveBasePermissions");
+        return new Gettable<any>(this);
+    }
+
+    /**
+     * Gets the effective base permissions for the item in a UI context
+     * 
+     */
+    public get effectiveBasePermissionsForUI(): Gettable<any> {
+        this.append("EffectiveBasePermissionsForUI");
+        return new Gettable<any>(this);
+    }
+
+    /**
+     * Gets the field values for this list item in their HTML representation
+     * 
+     */
+    public get fieldValuesAsHTML(): SelectableGettable<any> {
+        this.append("FieldValuesAsHTML");
+        return new SelectableGettable<any>(this);
```

Comment:
In some cases yes, we could build out data classes with all of the possible properties instead of any. For effective base permissions that may make sense. For example for fieldValuesAsHTML, how would you model that? A list can have any number of field with any number of names and value types. That would be very challenging to model. Better I think in the low level library to return any in that case and let the caller deal with what they get. Always open to ideas on that.